content/docs/functions/json_build_object.md

```diff
@@ -0,0 +1,265 @@
+---
+title: Postgres json_build_object function
+subtitle: "Builds a JSON object out of a variadic argument list."
+enableTableOfContents: true
+---
+
+
+Use `json_build_object` to construct a JSON object from a set of key-value pairs, creating a JSON representation of a row or set of rows.
+
+
+Function signature:
+
+
+```sql
+json_build_object ( VARIADIC "any" ) → json
+```
+
+
+## `json_build_object` example
+
+
+Let's consider a scenario where we have a table storing information about users:
+
+
+**users**
+
+```text
+| id |   name   | age |   city  
+|----|----------|-----|----------
+| 1  | John Doe |  30 | New York |
+| 2  | Jane Doe |  25 | London   |
+| 3  | John Doe |  30 | New York |
+| 4  | Jane Doe |  25 | London   |
+```
+
+
+{/*
+```sql
+CREATE TABLE users (
+ id SERIAL PRIMARY KEY,
+ name TEXT NOT NULL,
+ age INTEGER,
+ city TEXT
+);
+
+
+INSERT INTO users (name, age, city)
+VALUES ('John Doe', 30, 'New York'),
+      ('Jane Doe', 25, 'London');
+```
+*/}
+
+
+Use `json_build_object` to create a JSON structure with user information:
+
+
+
+
+```sql
+SELECT id,
+ json_build_object(
+   'name', name,
+   'age', age,
+   'city', city
+ ) AS user_data
+FROM users;
+```
+
+
+Returns:
+
+```text
+| id |                       user_data                       
+|----|--------------------------------------------------------
+| 1  | {"name" : "John Doe", "age" : 30, "city" : "New York"}
+| 2  | {"name" : "Jane Doe", "age" : 25, "city" : "London"}
+```
+
+
+## Advanced examples
+
+
+### Building nested JSON objects
+
+
+Let’s say we have a table of products with an `attributes` column containing JSON data:
+
+
+**products**
+
+```text
+| id |    name     | price |             description              | category |                           attributes                           
+|----|-------------|-------|-------------------------------------|----------|---------------------------------------------------------------
+| 1  | T-Shirt     | 25.99 | A comfortable cotton T-Shirt        | Clothing | {"size": "Medium", "color": "Blue"}
+| 2  | Coffee Mug  | 12.99 | A ceramic mug with a funny design   | Kitchen  | {"size": "Large", "color": "White", "material": "Ceramic"}
+```
+
+
+{/*
+```sql
```

Comment:
Why not just create one "products" table for this overall section and have it include the ratings? Instead of creating a second "products_with_ratings" table. That seems more straightforward, especially if someone wants to actually play with these queries.